backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,223 @@
+                    ? "urn:test"
+                    : input.RedirectUri,
+                    UriKind.Absolute)
+            },
+            PostLogoutRedirectUris =
+            {
+                new Uri(environment.IsEnvironment(Program.TestEnvironment)
+                    ? "urn:test"
+                    : input.PostLogoutRedirectUri,
+                    UriKind.Absolute)
+            },
+            Permissions =
+            {
+                // Add default permissions
+                OpenIddictConstants.Permissions.Endpoints.Authorization,
+                OpenIddictConstants.Permissions.Endpoints.Introspection,
+                OpenIddictConstants.Permissions.Endpoints.Logout,
+                OpenIddictConstants.Permissions.Endpoints.Revocation,
+                OpenIddictConstants.Permissions.Endpoints.Token,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.GrantTypes.Password
+                    : OpenIddictConstants.Permissions.GrantTypes.AuthorizationCode,
+                OpenIddictConstants.Permissions.GrantTypes.RefreshToken,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.ResponseTypes.Token
+                    : OpenIddictConstants.Permissions.ResponseTypes.Code,
+                OpenIddictConstants.Permissions.Scopes.Address,
+                OpenIddictConstants.Permissions.Scopes.Email,
+                OpenIddictConstants.Permissions.Scopes.Phone,
+                OpenIddictConstants.Permissions.Scopes.Profile,
+                OpenIddictConstants.Permissions.Scopes.Roles,
+            },
+            Requirements =
+            {
+                OpenIddictConstants.Requirements.Features.ProofKeyForCodeExchange
+            }
+        };
+
+        var permissions = JsonSerializer.Deserialize<List<string>>(input.Permissions);
+        permissions?.ForEach(permission => descriptor.Permissions.Add(permission));
+
+        return new CreateApplicationPayload(await applicationManager.CreateAsync(descriptor, cancellationToken).AsTask().ConfigureAwait(false));
+    }
+
+    [UseUserManager]
+    [Authorize(Policy = AuthConfiguration.WritePolicy)]
+    public async Task<UpdateApplicationPayload> UpdateApplicationAsync(
+        UpdateApplicationInput input,
+        OpenIddictApplicationManager<OpenIddictEntityFrameworkCoreApplication> applicationManager,
+        ClaimsPrincipal claimsPrincipal,
+        UserManager<User> userManager,
+        ApplicationDbContext context,
+        CancellationToken cancellationToken
+    )
+    {
+        if (input.Id == null)
+        {
+            return new UpdateApplicationPayload(
+                new UpdateApplicationError(UpdateApplicationErrorCode.UNKNOWN,
+                "Empty Id",
+                Array.Empty<string>()));
+        }
+        if (!await OpenIdConnectAuthorization.IsAuthorizedToManageApplications(
```

Comment:
Authorization should always happen before anything else, also before null-checks if possible.